### ISSUE TITLE
Error message: xxx_status

### ISSUE THREAD

**Issue by odow** [JuliaOpt/JuMP.jl#1618]
Also `primal_status` etc.
```julia
model = Model()
JuMP.dual_status(model)

julia> JuMP.dual_status(model)
ERROR: MethodError: no method matching get(::Nothing, ::MathOptInterface.DualStatus)
Closest candidates are:
  get(::Model, ::MathOptInterface.AbstractModelAttribute) at C:\Users\Oscar\.julia\packages\JuMP\t3yHD\src\JuMP.jl:453
  get(::Ipopt.Optimizer, ::MathOptInterface.DualStatus) at C:\Users\Oscar\.julia\packages\Ipopt\45sRy\src\MOIWrapper.jl:783
  get(::MathOptInterface.Utilities.MockOptimizer, ::MathOptInterface.DualStatus) at C:\Users\Oscar\.julia\packages\MathOptInterface\QO3Mk\src\Utilities\mockoptimizer.jl:162
  ...
Stacktrace:
 [1] get(::MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{JuMP.JuMPMOIModel{Float64}}}, ::MathOptInterface.DualStatus) at C:\Users\Oscar\.julia\packages\MathOptInterface\QO3Mk\src\Utilities\cachingoptimizer.jl:433
 [2] get(::MathOptInterface.Bridges.LazyBridgeOptimizer{MathOptInterface.Utilities.CachingOptimizer{MathOptInterface.AbstractOptimizer,MathOptInterface.Utilities.UniversalFallback{JuMP.JuMPMOIModel{Float64}}},MathOptInterface.Bridges.AllBridgedConstraints{Float64}}, ::MathOptInterface.DualStatus) at C:\Users\Oscar\.julia\packages\MathOptInterface\QO3Mk\src\Bridges\bridgeoptimizer.jl:172
 [3] get(::Model, ::MathOptInterface.DualStatus) at C:\Users\Oscar\.julia\packages\JuMP\t3yHD\src\JuMP.jl:453
 [4] dual_status(::Model) at C:\Users\Oscar\.julia\packages\JuMP\t3yHD\src\JuMP.jl:341
 [5] top-level scope at none:0
```

**Comment by odow** [comment]
@blegat I'm getting confused with all of the layers. Who has the responsibility for reporting that no one supports `TerminationStatus`? Is it sufficient (and best practice) to go:
```julia
# Internal function: used to check whether the model has a concrete backend.
function has_concrete_backend(model::Model)
    if JuMP.mode(model) == JuMP.Direct
        return true
    else
        cache = JuMP.caching_optimizer(model)
        return cache.optimizer !== nothing
    end
end

function termination_status(model::Model)
    if has_concrete_backend(model)
        return MOI.get(model, MOI.TerminationStatus())
    else
        error("Unable to get termination status because model has not been " *
              "solved.")
    end
end
```

**Comment by blegat** [comment]
The motivation of the layers is that you should not need to manually unwrap them. You can make all the calls from the top. Here it is a but in `CachingOptimizer`, it should return an appropriate `TerminationStatus`. This status should be the same than the status that a solver would return if you have setup and model but not called `MOI.optimize!` yet.
By the way I think we are missing such status. What about `MOI.Unsolved` ? We had `:Undecided` with MPB IIRC.

**Comment by odow** [comment]
I think `UnknownResultStatus` or `NoSolution` status is fine.

**Comment by blegat** [comment]
These are result status not termination status

**Comment by odow** [comment]
I've opened an issue. https://github.com/JuliaOpt/MathOptInterface.jl/issues/572

**Comment by odow** [comment]
Closing as this was fixed by MOI v0.6.4.
